### ISSUE TITLE
@chain one-liner nested in an outer @chain may not work?

### ISSUE THREAD

**Issue by max-sixty** [jkrumbiegel/Chain.jl#42]
This might be "overusing" `@chain`, but I found there was an inconsistency between one-liner and nested blocks with this:

This works:
```julia
@chain read(open("Day-8.txt"), String) begin
    strip
    split("\n")
    map(_) do _
        map(split(_, "|")) do _
            @chain _ begin
                strip
                split
            end
        end
    end
end
```

But making the inner `@chain` into a one-line version fails with "Malformed nested @chain macro"
```julia
@chain read(open("Day-8.txt"), String) begin
    strip
    split("\n")
    map(_) do _
        map(split(_, "|")) do _
            @chain _ strip split
        end
    end
end
```
FYI I tried using `do x` rather than `do _`, but to no avail.

I'm not sure the nesting is a good design — I was attempting to parse a heavily nested string, and was using new indentation to signify each level of nesting — so I'm putting this in for info, rather than something that's important per-se.

Thanks for the excellent package, I'm really enjoying using it

**Comment by max-sixty** [comment]
Tangentially — if there were a way of combining the `map ... do x` with `@chain`; that would be ideal; e.g.:

```julia
@chain read(open("Day-8.txt"), String) begin
    strip
    split("\n")
    @. begin
        split("|")
        @. begin
            strip
            split
        end
    end
end

```

**Comment by jkrumbiegel** [comment]
The single line nested chain error is a bug, fixed that here https://github.com/jkrumbiegel/Chain.jl/pull/43

The other thing is that you should not really repeat the underscore so often, it can work if the variable inside `map` just shadows the outer one, but it's not very clear. Better like this:

```julia
@chain "a\nb" begin
    strip
    split("\n")
    map(_) do x
        map(split(x, "|")) do y
            @chain y strip split
        end
    end
end
```

**Comment by jkrumbiegel** [comment]
Ah and I guess you don't need the underscore in the nested `@chain` anyway, because it's implicitly inserted

**Comment by max-sixty** [comment]
> The single line nested chain error is a bug, fixed that here #43

Thanks a lot!

I agree re the underscores. That's related to something I liked about the idea of combining the `map ... do x` with `@chain` (with, say `@.` as in the example, though maybe there are better names) — the whole block could be point-free.

**Comment by max-sixty** [comment]
Re the narrower issue — confirmed it works — thanks again. I'll close this issue
